### PR TITLE
[DebugInfo] Remove "_V_" from parameter names in case of byvalue parameters

### PR DESCRIPTION
**Pull request by SouraVX**
flang frontend prefixes "_V_" to the name of variables when parameters
are passed by value. These names are translated to LLVM-IR then to actual
object resulting in incorrect(from perspective of source level debugging)
debug-info.

This patch fixs this by replacing the symbol name to actual symbol name
at LLVM-IR translation(flang2) level. Benefit for this approach is that,
frontend(flang1) is unaware/un-touched by this patch.

For typical byvalue test case:
Incorrect LLVM-IR:
call void @llvm.dbg.declare(metadata i32* %_V_foo, metadata !9, metadata !DIExpression()), !dbg !11
Incorrect dwarf:
0x0000004c:       DW_TAG_formal_parameter
                    DW_AT_location      (DW_OP_fbreg +52)
                    DW_AT_name  ("_V_foo")
                    DW_AT_type  (0x00000090 "integer")

After this patch:
call void @llvm.dbg.declare(metadata i32* %foo, metadata !9, metadata !DIExpression()), !dbg !11
0x0000004c:       DW_TAG_formal_parameter
                    DW_AT_location      (DW_OP_fbreg +52)
                    DW_AT_name  ("foo")
                    DW_AT_type  (0x00000090 "integer")

### CODE REVIEW

File: tools/flang2/flang2exe/lldebug.cpp [hunk 1]
```diff
@@ -3110,8 +3110,9 @@ lldbg_emit_local_variable(LL_DebugInfo *db, SPTR sptr, int findex,
       flags = 0;
     }
     var_mdnode = lldbg_create_local_variable_mdnode(
-        db, DW_TAG_auto_variable, blk_info->mdnode, symname, file_mdnode, 0, 0,
-        type_mdnode, flags, fwd);
+        db, DW_TAG_auto_variable, blk_info->mdnode, PASSBYVALG(sptr)
+       ? SYMNAME(MIDNUMG(sptr)) : symname, file_mdnode, 0, 0,
+                type_mdnode, flags, fwd, sptr);
```
Comment (kiranchandramohan) [inline tools/flang2/flang2exe/lldebug.cpp:3115]: There seems to be more parameters here. Did you miss uploading the diff in the definition of lldbg_create_local_variable_mdnode or is there a typo here?

Comment (SouraVX) [reply tools/flang2/flang2exe/lldebug.cpp:3115]: Seems legit to me. Are you facing any problems while building this ?

Comment (kiranchandramohan) [reply tools/flang2/flang2exe/lldebug.cpp:3115]: Isn't sptr an additional parameter here?

Comment (kiranchandramohan) [reply tools/flang2/flang2exe/lldebug.cpp:3115]: Yes, I faced a build issue due to additional parameters.

Comment (SouraVX) [reply tools/flang2/flang2exe/lldebug.cpp:3115]: No. That may seem due to ternary operator inside this call.

Comment (SouraVX) [reply tools/flang2/flang2exe/lldebug.cpp:3115]: I checked it's building fine.

Comment (SouraVX) [reply tools/flang2/flang2exe/lldebug.cpp:3115]: Oops Thanks for this! I noticed Our down stream changes contain an extra `sptr`. Could you please put this PR review on hold.

**Comment by SouraVX** [comment]
Created a new one.

**Comment by SouraVX** [comment]
Created a new one https://github.com/flang-compiler/flang/pull/1106

**Pull request closed without merging**